### PR TITLE
Catch invalid query params

### PR DESCRIPTION
**Pull request by Lvl4Sword**
This way there isn't a server error when trying negative counts ( ?count=-1 for example )

### CODE REVIEW

File: Abe/abe.py [hunk 1]
```diff
@@ -415,13 +415,20 @@ def handle_chain(abe, page):
...
+        except ValueError:
             count = 20
-        hi = get_int_param(page, 'hi')
-        orig_hi = hi
+
+        try:
+            hi = get_int_param(page, 'hi')
+            orig_hi = hi
+        except ValueError:
+            hi = 19
```
Comment (dermoth) [inline Abe/abe.py]: Set to none here, let Abe find the tip

**Comment by dermoth** [comment]
My question was simple... what happens right now with a negative count? Shows 0 records or Exception? I'd like to know.

Regarding the new patch, I think `hi` should be set to `None` on ValueError, i.e. default behavior with no param.

Thanks

**Comment by Lvl4Sword** [comment]
Negative count would give the error:
`A server error occurred.  Please contact the administrator.`
I'll fix the `hi` parameter, one moment..

**Comment by Lvl4Sword** [comment]
That should work now :+1:

File: Abe/abe.py [hunk 2]
```diff
@@ -415,11 +415,20 @@ def handle_chain(abe, page):
...
+            elif count <= 0:
+                count = 20
+        except ValueError:
             count = 20
-        hi = get_int_param(page, 'hi')
-        orig_hi = hi
+
+        try:
+            hi = get_int_param(page, 'hi')
+            orig_hi = hi
```
Comment (dermoth) [inline Abe/abe.py]: I should have paid more attention on my first review... that orig_hi should go after the try block - looking at the original core it was always set to the same value as hi but now won't exist after exception handling.

**Comment by dermoth** [comment]
I squashed your commits as there were too many back and forth changes, wouldn't make much sense to someone else reviewing the commits and the squashed diff is quite straightforward.

This means you'll probably have to drop your patch-1 as it diverged now.

Thanks

**Pull request merged**